### ISSUE TITLE
Pre-built releases link against glibc and thus do not work on musl-based Linuxes

### ISSUE THREAD

**Issue by klardotsh** [synfinatic/aws-sso-cli#749]
tl;dr I don't know what all might break if you do this but `CGO_ENABLED=0` made a static build locally, or you can run an Alpine build in CI and publish a musl artifact (less ideal).

---

**Output of `aws-sso version`:** Impossible to collect because it doesn't run.

**Describe the bug:**
Title mostly says all. LDD proof:

```
$ ldd ~/Downloads/aws-sso-1.14.2-linux-amd64 
	/lib64/ld-linux-x86-64.so.2 (0x7fbe640f9000)
	libc.so.6 => /lib64/ld-linux-x86-64.so.2 (0x7fbe640f9000)
Error relocating /home/j/Downloads/aws-sso-1.14.2-linux-amd64: __vfprintf_chk: symbol not found
Error relocating /home/j/Downloads/aws-sso-1.14.2-linux-amd64: __fprintf_chk: symbol not found
```

**To Reproduce:**
 1. Download the amd64 Linux release binary on Alpine or Void Linux (musl variant)
 2. Attempt to use it
 3. Fail here

**Expected behavior:**
As an end user the expected behavior is that I can run the pre-built binary on Alpine, Void, whatever.

As a developer, I dug in and found that the issue is missing symbols, indicating a link against glibc. This surprised me, because Go binaries typically come out fully statically linked (this is how, for example, Terraform binaries work on any Linux).

Pulling the repo and running `CGO_ENABLED=0 make` produced a `dist/aws-sso` that was statically linked:

```
(snowcone) aws-sso-cli  » main » CGO_ENABLED=0 make
go build -trimpath -ldflags='-X "main.Version=1.14.2" -X "main.Delta=''" -X "main.Buildinfos=2024-01-04T15:23:56-0800" -X "main.Tag=v1.14.2-47-g7055d16" -X "main.CommitID=7055d16fad4e01a3aa0362bce296d7064489ca67"' -o dist/aws-sso ./cmd/aws-sso/...
Created: dist/aws-sso
                                                                                                         
(snowcone) aws-sso-cli  » main » ldd dist/aws-sso
ldd: dist/aws-sso: Not a valid dynamic program
                                                                                                         
(snowcone) aws-sso-cli  » main » dist/aws-sso 
WARNING No config file found!  Will now prompt you for a basic config... 

**********************************************************************
* Do you have questions?  Do you like reading docs?  We've got docs! *
* https://github.com/synfinatic/aws-sso-cli/blob/main/docs/config.md *
**********************************************************************

... etc
```

**Desktop (please complete the following information):**
```
$ uname -a && lsb_release -a && file /lib/libc.so 
Linux snowcone 6.6.8_1 #1 SMP PREEMPT_DYNAMIC Wed Dec 20 23:04:20 UTC 2023 x86_64 GNU/Linux
LSB Version:	1.0
Distributor ID:	VoidLinux
Description:	Void Linux
Release:	rolling
Codename:	void
/lib/libc.so: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), static-pie linked, BuildID[sha1]=389edf83097dc192a43656aedc87807cc18cc328, with debug_info, not stripped
```

**Comment by synfinatic** [comment]
Hi @klardotsh,

Yeah, CGO_ENABLED=1 is the Go default and I haven't overridden it.  And it's actually required on macOS to support Keychain.   I know disabling CGO changes how DNS resolution works and in theory it should work, but I'm a little afraid of some issues because people often don't really grok systemd.

Anyways, is this request coming from the perspective of wanting to run the ecs-server in a container?   Or are you actually running aws-sso interactively from a musl based distro?

**Comment by klardotsh** [comment]
I run Void Linux (musl variant) on all of my personal (interactive, daily driver) workstations - I'm one of those bizarre niche usecases that sucks to support, admittedly 😄

Bonus context: I'm using https://github.com/adamcrews/asdf-aws-sso-cli to roll this tool out to all developers on my team - mind you, they're mostly on MacOS so it's N/A for a lot of them, and the rest are on glibc Linuxes. That said, there's [an open issue](https://github.com/adamcrews/asdf-aws-sso-cli/issues/6) in that repo about broken dynamic linking, so I think there's maybe something to be gained by offering static builds - even if they're offered as alternatives to the glibc-linked stuff.

Anyway, I hacked around it by installing a static copy to `~/.local/bin/aws-sso` and letting the mise/asdf installed copy fail (and thus fallback to the next match in `$PATH`, since the mise-installed copy is "not found" as a valid binary). Not ideal, and adds a corner case to my README, but it works.

If you'd rather not deal with this upstream (which I understand), maybe I'll hack up a fork of the `asdf` provider that checks the validity of the installed binary and on failure rebuilds from source? Kinda hacky, but I think any solution in this space will be.

**Comment by synfinatic** [comment]
Ah, got it.  Yeah, never heard of a musl based user-space linux before... that sounds uh, "interesting". :D  I think this is doable.

**Comment by synfinatic** [comment]
@klardotsh btw, you should be able to pull down main, run `make linux` and give it a go.  LMK if you have any other issues
